### PR TITLE
Fix AnchorNav focus bug

### PR DESCRIPTION
**Pull request by joshfarrant**
## Summary

Resolves an issue where the focus was trapped on mobile when using the `AnchorNav` component. This PR adds a new attribute to the `span` element in the `AnchorNav` component to allow the focus to move forward when the tab key is pressed.

## List of notable changes:

- `data-forward-focus` attribute added to span which wraps `AnchorNav` actions. The name is arbitrary, but it is used to determine if the focus should instead be forwarded to the element's first child
- Fixes a few minor typos in the `AnchorNav` test file
- DOES NOT add any tests for the focus fix. I couldn't get tabbing and focus tracking to work correctly inside the unit tests unfortunately. 

## What should reviewers focus on?

- Are you happy with the `data-forward-focus` approach?

## Steps to test:

- Check that you're able to tab from any link in the `AnchorNav` to the first of the actions

## Supporting resources (related issues, external links, etc):

- #608 

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

**Review by rezrah — APPROVED** [review]
Fix works as expected 👍

Just spotted a separate issue around focus trapping incorrectly persisting when mobile resizes back up to desktop, but this is a wider issue with the hook, which doesn't need to block this from rolling out.

@joshfarrant - before merging, could you please add a changeset to this PR, and bring the branch up-to-date with main

**Comment by joshfarrant** [comment]
Thanks @rezrah!

I always forget the changesets 🤦‍♂️

**Pull request merged**
